src/redis_cmd.cc

```diff
@@ -4750,6 +4756,883 @@ class CommandScript : public Commander {
+    bool entry_id_found = false;
+    stream_name_ = args[1];
+
+    for (size_t i = 2; i < args.size(); ) {
+      auto val = Util::ToLower(args[i]);
+
+      if (val == "nomkstream" && !entry_id_found) {
+        nomkstream_ = true;
+        ++i;
+        continue;
+      }
+
+      if (val == "maxlen" && !entry_id_found) {
+        if (i+1 >= args.size()) {
+          return Status(Status::RedisParseErr, errInvalidSyntax);
+        }
+
+        size_t max_len_idx;
+        bool eq_sign_found = false;
+        if (args[i+1] == "=") {
+          max_len_idx = i+2;
+          eq_sign_found = true;
+        } else {
+          max_len_idx = i+1;
+        }
+
+        if (max_len_idx >= args.size()) {
+          return Status(Status::RedisParseErr, errInvalidSyntax);
+        }
+
+        try {
+          max_len_ = std::stoull(args[max_len_idx]);
+          with_max_len_ = true;
+        } catch (const std::exception &) {
+          return Status(Status::RedisParseErr, errValueNotInterger);
+        }
+
+        i += eq_sign_found ? 3 : 2;
+        continue;
+      }
+
+      if (val == "minid" && !entry_id_found) {
+        if (i+1 >= args.size()) {
+          return Status(Status::RedisParseErr, errInvalidSyntax);
+        }
+
+        size_t min_id_idx;
+        bool eq_sign_found = false;
+        if (args[i+1] == "=") {
+          min_id_idx = i+2;
+          eq_sign_found = true;
+        } else {
+          min_id_idx = i+1;
+        }
+
+        if (min_id_idx >= args.size()) {
+          return Status(Status::RedisParseErr, errInvalidSyntax);
+        }
+
+        try {
+          auto s = ParseStreamEntryID(args[min_id_idx], &min_id_);
+          if (!s.IsOK()) {
+            return Status(Status::RedisParseErr, s.Msg());
+          }
+          with_min_id_ = true;
+        } catch (const std::exception &) {
+          return Status(Status::RedisParseErr, errValueNotInterger);
+        }
+
+        i += eq_sign_found ? 3 : 2;
+        continue;
+      }
+
+      if (val == "limit" && !entry_id_found) {
+        return Status(Status::RedisParseErr, errLimitOptionNotAllowed);
+      }
+
+      if (val == "*" && !entry_id_found) {
+        entry_id_found = true;
+        ++i;
+        continue;
+      } else if (!entry_id_found) {
+        auto s = ParseNewStreamEntryID(val, &entry_id_);
+        if (!s.IsOK()) {
+          return Status(Status::RedisParseErr, s.Msg());
+        }
+        entry_id_found = true;
+        with_entry_id_ = true;
+        ++i;
+        continue;
+      }
+
+      if (entry_id_found) {
+        name_value_pairs_.push_back(val);
+        ++i;
+      }
+    }
+
+    if (name_value_pairs_.empty() || name_value_pairs_.size() % 2 != 0) {
+      return Status(Status::RedisParseErr, errWrongNumOfArguments);
+    }
+
+    return Status::OK();
+  }
+
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    if (svr->IsSlave()) {
```

Comment:
Can remove this check since we have already done it at: https://github.com/apache/incubator-kvrocks/blob/40132067bc7bb696f96675e8a396f0473321a454/src/redis_connection.cc#L407